browser/ui/views/split_view/split_view.h

```diff
@@ -0,0 +1,139 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_UI_VIEWS_SPLIT_VIEW_SPLIT_VIEW_H_
+#define BRAVE_BROWSER_UI_VIEWS_SPLIT_VIEW_SPLIT_VIEW_H_
+
+#include <memory>
+#include <vector>
+
+#include "base/functional/callback_forward.h"
+#include "base/memory/weak_ptr.h"
+#include "base/types/pass_key.h"
+#include "brave/browser/ui/tabs/split_view_browser_data.h"
+#include "brave/browser/ui/tabs/split_view_browser_data_observer.h"
+#include "brave/browser/ui/views/split_view/split_view_layout_manager.h"
+#include "chrome/browser/ui/tabs/tab_model.h"
+#include "ui/base/metadata/metadata_header_macros.h"
+#include "ui/views/view.h"
+
+namespace content {
+class WebContents;
+}  // namespace content
+
+namespace views {
+class WebView;
+}  // namespace views
+
+class BraveBrowserView;
+class Browser;
+class ContentsWebView;
+class DevToolsContentsResizingStrategy;
+class SplitViewLayoutManager;
+class SplitViewLocationBar;
+class SplitViewSeparator;
+
+// Contains a pair of contents container view.
+class SplitView : public views::View, public SplitViewBrowserDataObserver {
+  METADATA_HEADER(SplitView, views::View)
+ public:
+  using BrowserViewKey = base::PassKey<BraveBrowserView>;
+
+  SplitView(Browser* browser,
```

Comment:
could we take a ref to `Browser` instead? We should never create this class without one, right?


```suggestion
  SplitView(Browser& browser,
```

Reply:
Yup, it could be ref. Done.